docs/reference/search/search-your-data/run-knn-search.asciidoc

```diff
@@ -0,0 +1,248 @@
+[[run-knn-search]]
+== Run a k-nearest neighbor (kNN) search
+++++
+<titleabbrev>Run a kNN search</titleabbrev>
+++++
+
+experimental::[]
+
+//tag::knn-def[]
+A _k-nearest neighbor_ (kNN) search finds the _k_ nearest vectors to a query
+vector, as measured by a similarity metric.
+//end::knn-def[]
+
+image::images/kNN/knn_diagram.svg[Approximate kNN search as a two-dimensional graph,align="center"]
+
+Common use cases for kNN include:
+
+* Product recommendations and recommendation engines
+* Similarity search for images or videos
+* Outlier detection and classification
+
+[discrete]
+[[knn-prereqs]]
+=== Prerequisites
+
+* To run a kNN search, you must be able to convert your data into meaningful
+<<dense-vector,`dense_vector`>> values, called embeddings. This typically
+requires the setup and training of a machine learning model, called an embedding
+model. Elastic doesn't currently offer any out-of-the-box models or solutions
+for creating embeddings.
++
+The fit of your embedding model to your data and use case has a significant
```

Comment:
May be can drop this sentence? First not sure about "embedding models". Also with our knn search solution we are not responsible for fitting, we assess the quality of knn search based on recall.

Reply:
Good point. I dropped it in https://github.com/elastic/elasticsearch/pull/80857/commits/29d37f194d53c1da67de4b7376da9939803b3b08.